zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -816,3 +816,250 @@ impl<F: FieldExt, const NUM_BYTES: usize> MaxGadget<F, NUM_BYTES> {
         Ok(select::value(lt, rhs, lhs))
     }
 }
+#[derive(Clone, Debug)]
+pub(crate) struct DivWordsGadget<F> {
+    dividend: util::Word<F>,
+    divisor: util::Word<F>,
+    quotient: util::Word<F>,
+    remainder: [Cell<F>; 32],
+    v0: [Cell<F>; 9],
+    lt_lo: LtGadget<F, 16>,
+    comparison_hi: ComparisonGadget<F, 16>,
+}
+
+impl<F: FieldExt> DivWordsGadget<F> {
+    pub(crate) fn construct(
+        cb: &mut ConstraintBuilder<F>,
+        dividend: util::Word<F>,
+        divisor: util::Word<F>,
+    ) -> Self {
+        let quotient = cb.query_word();
+        let remainder = cb.query_bytes();
+        let v0 = cb.query_bytes();
+
+        let lt_lo = LtGadget::construct(
+            cb,
+            from_bytes::expr(&remainder[0..16]),
+            from_bytes::expr(&divisor.cells[0..16]),
+        );
+        let comparison_hi = ComparisonGadget::construct(
+            cb,
+            from_bytes::expr(&remainder[16..32]),
+            from_bytes::expr(&divisor.cells[16..32]),
+        );
+        let (lt_hi, eq_hi) = comparison_hi.expr();
+        let lt = select::expr(lt_hi, 1.expr(), eq_hi * lt_lo.expr());
+        cb.require_equal("remainder < quotient", 1.expr(), lt);
+
+        let mut dividend_limbs = vec![];
+        let mut divisor_limbs = vec![];
+        let mut quotient_limbs = vec![];
+        let mut remainder_limbs = vec![];
```

Comment:
You can just `remainder_hi` and `remainder_lo`, no need for 4 limbs

Reply:
Also `remainder_hi`, `remainder_lo` can be reused in the comparison and lt gadget above.